File: frontend/src/types/assignment.ts
import type { AssetStatus } from './asset'

export interface Assignment {
  id: string
  assetId: string
  assetCode: string
  assetName: string
  assetCategory: string
  assetBrand: string
  assetModel: string
  assetStatus: AssetStatus
  employeeId: string
  employeeNo: string
  employeeName: string
  department: string
  assignedAt: string
  returnedAt: string | null
  assignedByUserId: string
  assignedByName: string
  returnedByUserId: string | null
  returnedByName: string | null
  notes: string | null
  returnNotes: string | null
  isActive: boolean
}

export interface Employee {
  id: string
  employeeNo: string
  fullName: string
  department: string
  email: string
}

export type AssignmentStatus = 'Aktif' | 'İade Edildi'

export interface CreateAssignmentInput {
  assetId: string
  employeeId: string
  assignedAt: string
  notes: string | null
}

export interface ReturnAssignmentInput {
  returnedAt: string
  returnNotes: string | null
}

File: backend/Entities/Assignment.cs
namespace TakipProgrami.Api.Entities;

public sealed class Assignment
{
    public string Id { get; set; } = string.Empty;
    public string AssetId { get; set; } = string.Empty;
    public string EmployeeId { get; set; } = string.Empty;
    public DateTimeOffset AssignedAt { get; set; }
    public DateTimeOffset? ReturnedAt { get; set; }
    public string AssignedByUserId { get; set; } = string.Empty;
    public string? ReturnedByUserId { get; set; }
    public string? Notes { get; set; }
    public string? ReturnNotes { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public Asset Asset { get; set; } = null!;
    public Employee Employee { get; set; } = null!;
    public AppUser AssignedByUser { get; set; } = null!;
    public AppUser? ReturnedByUser { get; set; }
}
